pkg/utils/http_server_test.go

```
@@ -0,0 +1,390 @@
+			Serve: func() error {
+				close(serveStarted)
+				return server.Serve(listener)
+			},
+		})
+	}()
+	<-serveStarted
+
+	client := &http.Client{Timeout: time.Second}
+	responseResult := make(chan error, 1)
+	go func() {
+		response, err := client.Get("http://" + listener.Addr().String())
```

golangci-lint (2.12.2)

[error] (*net/http.Client).Get must not be called. use (*net/http.Client).Do(*http.Request)

(noctx)